pglookout/webserver.py

```diff
@@ -17,43 +22,62 @@
 
 
 class ThreadedWebServer(ThreadingMixIn, HTTPServer):
-    cluster_state = None
-    log = None
-    cluster_monitor_check_queue = None
-    allow_reuse_address = True
+    allow_reuse_address: bool = True
+
+    def __init__(
+        self,
+        address: str,
+        port: int,
+        RequestHandlerClass: type[BaseHTTPRequestHandler],
+        cluster_state: dict[str, MemberState],
+        log: Logger,
+        cluster_monitor_check_queue: Queue[str],
+    ) -> None:
+        super().__init__((address, port), RequestHandlerClass)
+        self.cluster_state: dict[str, MemberState] = cluster_state
+        self.log: Logger = log
+        self.cluster_monitor_check_queue: Queue[str] = cluster_monitor_check_queue
```

Comment:
I'd not mix this change with typing changes. And, this comment should be in the commit message anyway.

Reply:
Sure, the commit message could be improved.
The type hint could not be done without doing the changes. Or that would have implied a lot of casts and asserts to help mypy figuring out that things were set at runtime. Instead, I decided to have it in a more classic class code style, because there was nothing that justified having this kind of syntax, relying on variable shadowing and post-initialization delegated to the caller.